app/src/main/java/com/infomaniak/mail/utils/Extensions.kt

```diff
@@ -140,6 +143,25 @@ fun Fragment.safeNavigateToNewMessageActivity(draftMode: DraftMode, messageUid:
 
 fun List<Message>.getLastMessageToExecuteAction(): Message = lastOrNull { !it.isDraft } ?: last()
 
+fun Uri.getFileNameAndSize(context: Context): Pair<String, Int>? {
+    return runCatching {
+        val projection = arrayOf(OpenableColumns.DISPLAY_NAME, OpenableColumns.SIZE)
+        context.contentResolver.query(this, projection, null, null, null)?.use { cursor ->
+            if (cursor.moveToFirst()) {
+                val displayName = cursor.getColumnIndexOrThrow(OpenableColumns.DISPLAY_NAME).let(cursor::getString)
+                val size = cursor.getColumnIndexOrThrow(OpenableColumns.SIZE).let(cursor::getInt)
+                displayName to size
+            } else {
+                Sentry.captureException(Exception("$this has empty cursor"))
+                null
+            }
+        }
+    }.getOrElse { exception ->
+        Sentry.captureException(exception)
+        null
+    }
+}
+
```

Comment:
From experience with kDrive, the `DISPLAY_NAME` and `SIZE` can be null depending on the phones and applications that created them.
It's a bad thing to rewrite code, which could be shared with kDrive in Core
https://github.com/Infomaniak/android-kDrive/blob/43357f5e2a9fbe2c47ea644ec9d36a92067fa15c/app/src/main/java/com/infomaniak/drive/utils/SyncUtils.kt#L47